python/heading_controller.py

```diff
@@ -0,0 +1,80 @@
+from control_modes import ControlModes
+from controller_selector import ControllerSelector
+from sailbot_msg.msg import Sensors
+
+# TODO: Add class methods to get error and feedback gain
+
+
+class HeadingController:
```

Comment:
It seems like the "headingController" class is the main class we will use and the "ControllerSelector" class is a helper class.  A general comment is that methods that are closely facing users (in our case the main boat_controller_node.py loop) should be well tested if possible, so I'm thinking if we can update our code to better utilize the "controller_selector.py" tests to test the `switchControMode` method instead